modules/ctrl_dbus/ctrl_dbus.c

```diff
@@ -152,19 +153,29 @@ on_handle_invoke(DBusBaresip *interface,
 {
 	char *response;
 	struct ctrl_st *st = arg;
-	char buf[8] = {0,0,0,0,0,0,0,1};
+	char buf[1] = {1};
+	size_t n;
 
 	str_dup(&st->command, command);
 
 	pthread_mutex_lock(&st->wait.mutex);
-	write(st->fd[1], buf, sizeof(buf));
-	pthread_cond_wait(&st->wait.cond, &st->wait.mutex);
+	n = write(st->fd[1], buf, sizeof(buf));
+	if (n == 1)
+		pthread_cond_wait(&st->wait.cond, &st->wait.mutex);
+
 	pthread_mutex_unlock(&st->wait.mutex);
 
-	mbuf_strdup(st->mb, &response, mbuf_get_left(st->mb));
-	dbus_baresip_complete_invoke(interface, invocation, response);
-	mem_deref(response);
-	st->mb = mem_deref(st->mb);
+	if (st->mb) {
+		mbuf_strdup(st->mb, &response, mbuf_get_left(st->mb));
```

Comment:
also check return value here ...

Reply:
ok. Will print a warning if `mbuf_strdup()` fails and return an empty string for dbus call.